_posts/2019-03-05-build-firefox-extension-with-ringcentral-embeddable.md

```diff
@@ -0,0 +1,304 @@
+---
+layout: post
+title:  "Building a Firefox add-on with RingCentral Embeddable"
+date:   2019-03-05 10:00:00 +0800
+categories: blog
+author: Embbnux Ji
+---
+
+![ringcentral-embeddable-with-firefox](https://user-images.githubusercontent.com/7036536/55455660-0e8f0000-5617-11e9-89be-17c008c7d3f4.png)
+
+In our previous [article](https://medium.com/ringcentral-developers/build-a-chrome-extension-with-ringcentral-embeddable-bb6faee808a3), we introduce how to use RingCentral Embeddable to build a Chrome extension to integration RingCentral voices and text features into any CRM website. Now RingCentral Embeddable has also supported Firefox. So now we can also use Firefox add-on to integrate RingCentral into any CRM website.
+
+In this article, we will build RingCentral Embeddable with Google API integration in Firefox add-on.
+
+**Related Link:**
+
+* Core widget: [RingCentral Embeddable](https://github.com/ringcentral/ringcentral-embeddable)
+* Full codes: [RingCentral Embeddable for Google in Firefox add-on](https://github.com/ringcentral/ringcentral-embeddable)
+* Base Firefox add-on: [RingCentral Embeddable Firefox extension template](https://github.com/embbnux/ringcentral-embeddable-firefox-extension)
+
+## Build Firefox add-on manifest.json
+
+Before we build a Firefox add-on, we need to define `manifest.json`
+
+```json
+{
+  "name": "RingCentral Embeddable for Google",
+  "description": "RingCentral Embeddable For Google with Firefox extension",
+  "version": "0.0.3",
+  "permissions": [
+    "http://*/",
+    "https://*/",
+    "storage",
+    "activeTab",
+    "tabs",
+    "identity",
+    "notifications",
+    "unlimitedStorage",
+    "https://*.google.com/*"
+  ],
+  "content_scripts": [
+    {
+      "matches": ["https://www.google.com/contacts/*", "https://contacts.google.com/*", "https://calendar.google.com/*", "https://mail.google.com/*"],
+      "js": [
+        "vendors/libphonenumber-js.min.js",
+        "clickToDial.js",
+        "content.js"
+      ],
+      "css": [
+        "clickToDial.css"
+      ]
+    }
+  ],
+  "icons": {
+    "16": "rc16.png",
+    "32": "rc32.png",
+    "48": "rc48.png",
+    "128": "rc128.png"
+  },
+  "background": {
+    "scripts": ["googleClient.js", "background.js"]
+  },
+  "browser_action": {
+    "default_icon": {
+      "16": "rc16.png",
+      "32": "rc32.png"
+    }
+  },
+  "content_security_policy": "script-src 'self' https://ringcentral.github.io/ringcentral-embeddable; object-src 'self'",
+  "manifest_version": 2,
+  "applications": {
+    "gecko": {
+      "id": "integration-embeddable-firefox@ringcentral.com",
+      "strict_min_version": "60.0"
+    }
+  }
+}
+```
+
+Firefox add-ons `manifest.json` file is very similar as chrome extension. We need to add permissions , content , background and content_security_policy . We will inject JS and CSS files into google related website, so we need to add website url into content_scripts.matches.
```

Comment:
`The 'manifest.json' file of Firefox addons ` is very similar `to` chrome `extensions`. We need to add `permissions, content, background,` and content_security_policy `configurations`. We will `be injecting` JS and CSS... website, so we need to add `urls of these websites` into `the` content_script.matches `list`.